### ISSUE TITLE
Feature Request: Guide Lines

### ISSUE THREAD

**Issue by jmickle66666666** [sabresaurus/SabreCSG#102]
<img width="659" alt="screenshot 2018-05-02 21 50 00" src="https://user-images.githubusercontent.com/7083717/39545926-ff60911a-4e52-11e8-98f0-d337852acd0c.png">

These red lines along the edges of a brush when you are moving it in Trenchbroom are horrendously useful when aligning geometry. Would be awesome to get them in SabreCSG!

**Comment by Henry00IS** [comment]
Is there a line for every edge on the brush or just the bounding box? Wouldn't staircases, complex groups or 2DSE brushes have thousands of lines, possibly obstructing the view more than it would help? I love the idea and it should be fairly straightforward to add but I would love to see a screenshot showing how it handles more complex brushes in case I am missing something obvious or misjudging this completely. :grin:

I am also wondering what happens when you have a displacement/terrain brush, will the guide lines move all over the place because the hills point in all sorts of directions? Or is it just the bounding box in all these cases?

**Comment by jmickle66666666** [comment]
<img width="612" alt="screenshot 2018-05-08 00 25 18" src="https://user-images.githubusercontent.com/7083717/39728253-672627cc-5256-11e8-9f12-6c2397300f93.png">
just the bounding box!

**Comment by Henry00IS** [comment]
Here you go, please try out [this branch with experimental guide lines](https://github.com/Henry00IS/SabreCSG/tree/GuideLines) and let me know what you think.

![image](https://user-images.githubusercontent.com/7905726/39778499-caef1e2a-5306-11e8-883e-4753e94c652d.png)

**Comment by Henry00IS** [comment]
I just pushed an update that fades the lines away like in your image.

![image](https://user-images.githubusercontent.com/7905726/39779645-22809396-530a-11e8-9d7b-36f2ed6e803a.png)

**Comment by jmickle66666666** [comment]
Feels very nice! It'd be good to get the guides in the Translate and Rotate controls too. 

Also I tend to work without Auto Rebuild enabled for the most part (because of performance issues), and since the guides don't clip with the transparent shader used for the brushes before build it isn't too helpful there.

**Comment by Henry00IS** [comment]
The brush bounds are now visible in translate and rotate mode as well, not just the bounds tool itself.

Give me a moment to check out your specific issue next.

**Comment by Henry00IS** [comment]
I am sorry to say that I don't know how to have the lines disappear behind the transparent brushes, I assume there's probably a shader queue trick but I can't figure it out. If this really turns into a problem for you then feel free to open up another issue later once the PR is merged and we'll put an up-for-grabs on it.

**Comment by jmickle66666666** [comment]
yeah no worries, I might look into it myself at some point

**Comment by jmickle66666666** [comment]
thank you so much for this BTW! it'll help a ton :)

**Comment by Henry00IS** [comment]
I think this turned out amazing, I took level designer Constaneer's thumbs up as an approval. Thanks for this suggestion, it's a fantastic addition to SabreCSG and I imagine this will be useful for a lot of people that have perspective issues attempting to align things. 😁

**Comment by Henry00IS** [comment]
In fact I find this works extremely well for 2D orthographic editing as well if you are into such things. The colors start to merge when you rotate the brush giving you a real clue what you are doing.

**Comment by Constaneer** [comment]
This will defiantly help out a lot! 👌

**Comment by jmickle66666666** [comment]
Yeah its something often brought up about trenchbroom and thought it would be a simple(ish) addition to help out a lot! Thanks again!
